misc/systemd/ipfs-api.socket

```diff
@@ -0,0 +1,15 @@
+# Enabling this will *completely override* any API listeners configured in your
+# config.
+
+[Unit]
+Description=IPFS API Socket
+
+[Socket]
+Service=ipfs.service
+FileDescriptorName=io.ipfs.api
+BindIPv6Only=true
+ListenStream=127.0.0.1:5001
```

Comment:
It would be great if the port values were retrieved from the `~/.ipfs/config` or even better `~/.config/ipfs/config` or some suitable [configuration directory](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#RuntimeDirectory=). But whatever the solution, this will be confusing for users, i.e. enable systemd sockets, at some point in the future want to change port values, update ~/.ipfs/config, does nothing, restart sockets, still does nothing, google it/remembers to change in systemd files.

Reply:
Yeah but there's really nothing we can do about that. systemd doesn't let one read a config file.

Well... technically I can write a generator and *generate* these socket files dynamically. However, that probably isn't worth it.

---

We could, alternatively, listen on the union of addresses.

Reply:
You could use an [`EnvironmentFile`](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#EnvironmentFile=)?